connection_params.ts

```diff
@@ -17,106 +18,117 @@ function getPgEnv(): IConnectionParams {
   }
 }
 
-function selectFrom(
-  sources: Array<IConnectionParams>,
-  key: keyof IConnectionParams,
-): string | undefined {
-  for (const source of sources) {
-    if (source[key]) {
-      return source[key];
-    }
-  }
-
-  return undefined;
+function isDefined<T>(value: T): value is Exclude<T, undefined | null> {
+  return value !== undefined && value !== null;
 }
 
-function selectFromWithDefault(
-  sources: Array<IConnectionParams>,
-  key: keyof typeof DEFAULT_CONNECTION_PARAMS,
-): string {
-  return selectFrom(sources, key) || DEFAULT_CONNECTION_PARAMS[key];
+class ConnectionParamsError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "ConnectionParamsError";
+  }
 }
 
-const DEFAULT_CONNECTION_PARAMS = {
-  host: "127.0.0.1",
-  port: "5432",
-  application_name: "deno_postgres",
-};
-
-export interface IConnectionParams {
+export interface ConnectionOptions {
   database?: string;
-  host?: string;
-  port?: string;
+  hostname?: string;
+  port?: number;
   user?: string;
   password?: string;
   application_name?: string;
```

Comment:
This field should be camelCased, could you do that @lenkan ?

Reply:
:+1: I renamed this one to `applicationName`.

Should I also rename the internally used `ConnectionParams.application_name` to camelCase? 

Also, what about the dsn-style parameters?

```
postgres://server/dbname?application_name=appName
```

or 


```
postgres://server/dbname?applicationName=appName
```

Reply:
DSN should behave as-is; ie. keep the snake_case, I'm fine with changing all internal usages to camelCase as well 👍

Reply:
Great! Fixed it.